packages/components/src/VirtualizedList/RowLarge/RowLarge.component.js

```diff
@@ -99,13 +128,19 @@ class RowLarge extends React.Component {
 				aria-label={titleField && getCellData(titleField, parent, index)}
 			>
 				<div className={`tc-list-large-inner-box ${theme['inner-box']}`} key="inner-box">
-					<div className={theme.header} key="header">
-						{titleCell}
-						{selectionCell}
-					</div>
-					<dl className={`tc-list-large-content ${theme.content}`} key="content">
-						{otherFields.map(this.renderKeyValue)}
-					</dl>
+					{isEmpty(rowData) ? (
+						<LargeInnerRowLoading columns={3} rows={3} />
```

Comment:
You can guess the rows number with otherFields.
You can keep 3 cols by default, then the number of row is otherFields / 3 (be careful 10 other fields = 4 row)

But I think a safer way to do that would be to apply the same css as with data, and you just generate a skeleton for each `otherFields`

Reply:
After looking at the current state of the code, I think it's a cleaner and simpler solution to compute the number of rows instead of using current rendering mechanism to show skeletons when we have a row with no data.